### PR TITLE
Fix CHANGES.txt for 8.7

### PR DESCRIPTION
**Pull request by atris**
Fixing merge issues with 8.7 change notes

### CODE REVIEW

**Comment by munendrasn** [comment]
Some change log entries which are meant only for master (due to backward compatibility break) has been backported to branch_8x in this commit https://github.com/apache/lucene-solr/commit/c841f63e4ba8615d648f00741d4ee3628cd60954
I think we would need to fix that too.

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -177,6 +77,15 @@ Improvements
...
 
+* SOLR-14223: PKI Auth can bootstrap from existing key files instead of creating new keys on startup (Mike Drob)
+
+* SOLR-11725: Use corrected sample formula for computing stdDev and variance in JSON aggregations
+  (hossman, Munendra S N, yonik)
+
+* SOLR-14387: SolrClient.getById() will escape comma separater within ids (Markus Schuch via Mike Drob)
+
+* SOLR-10814: Add short-name feature to RuleBasedAuthz plugin (Mike Drob, Hrishikesh Gadre)
+
```
Comment (munendrasn) [inline solr/CHANGES.txt:88]: AFAIK, this shouldn't be in either 8x or 8_7 branches. These are master only changes

Comment (munendrasn) [reply solr/CHANGES.txt:88]: I have created PR #2033 which fixes changelog for branch_8x. If it looks good then, we can port it to branc_8_7

Comment (munendrasn) [reply solr/CHANGES.txt:88]: Also, there are entries like below
```
* SOLR-14859: DateRangeField now throws errors when invalid field/fieldType options specified; no longer silently accepts incompatible option values
  (Jason Gerlowski, Chris Hostetter, Munendra S N)
```
This was released in 8.6.3, not sure if it should be again included in 8.7 release

Comment (jpountz) [reply solr/CHANGES.txt:88]: Based on our release process, changes entries for released versions should not appear in the changes of unreleased versions, so we should drop this one indeed. See https://cwiki.apache.org/confluence/display/LUCENE/ReleaseTodo#ReleaseTodo-SynchronizeCHANGES.txt.

**Pull request closed without merging**